cli/flox/src/commands/environment.rs

```diff
@@ -1202,26 +1202,54 @@ impl Update {
     }
 }
 
-/// upgrade packages using their most recent flake
+/// Upgrade packages in an environment
+///
+/// When no arguments are specified, all packages in the environment are
+/// upgraded.
+///
+/// Packages to upgrade can be specified by either group name or, if a package
+/// is not in a group with any other packages, it may be specified by ID.
+///
+/// If the specified argument is both a group name and a package ID, only the
+/// group is upgraded.
+///
+/// Packages without a specified group in the manifest can be upgraded by
+/// passing 'toplevel' as the group name.
```

Comment:
You can add a couple more newlines in the doc to force a newline

```suggestion
/// Upgrade packages in an environment
///
///
///
///
/// When no arguments are specified, all packages in the environment are
/// upgraded.
///
/// Packages to upgrade can be specified by either group name or, if a package
/// is not in a group with any other packages, it may be specified by ID.
///
/// If the specified argument is both a group name and a package ID, only the
/// group is upgraded.
///
/// Packages without a specified group in the manifest can be upgraded by
/// passing 'toplevel' as the group name.
```

Produces: 

```
    Upgrade packages in an environment
    
      When no arguments are specified, all packages in the environment are upgraded.
      Packages to upgrade can be specified by either group name or, if a package is not in a group with any
      other packages, it may be specified by ID.
      If the specified argument is both a group name and a package ID, only the group is upgraded.
      Packages without a specified group in the manifest can be upgraded by passing 'toplevel' as the group
      name.
```

As we put in more description we also might want to use the more explcit parser attributes:

```rust
#[bpaf(command, descr("descr"), header("header"), footer("footer"), help("help"))]
struct Upgrade
...
```

Reply:
I had to move things to the command enum, and now I'm getting:
```
> flox upgrade --help
Upgrade packages in an environment

Usage: flox upgrade [-s=SYSTEM] [-d=<path> | -r=<owner/name>] [<package or group>]...

When no arguments are specified, all packages in the environment are upgraded.

Packages to upgrade can be specified by either group name or, if a package is not in a group with any
other packages, it may be specified by ID. If the specified argument is both a group name and a package
ID, only the group is upgraded.

Packages without a specified group in the manifest can be upgraded by passing 'toplevel' as the group
name.

Environment Options
    -s, --system=SYSTEM

Available positional items:
    <package or group>   ID of a package or group name to upgrade

Available options:
    -d, --dir=<path>     Path containing a .flox/ directory
    -r, --remote=<owner/name>  A remote environment on floxhub
    -h, --help           Prints help information
```

Reply:
Yeah that's the alternative, this just makes it being a bit detached from the argument declaration

Reply:
I think that's better than the first paragraph not being aligned with everything afterwards